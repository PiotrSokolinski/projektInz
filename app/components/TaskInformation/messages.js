/*
 * TaskInformation Messages
 *
 * This contains all the text for the TaskInformation component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.TaskInformation'

export default defineMessages({
  description: {
    id: `${scope}.description`,
    defaultMessage: 'Description',
  },
  author: {
    id: `${scope}.author`,
    defaultMessage: 'Author: {author}',
  },
  createdAt: {
    id: `${scope}.date`,
    defaultMessage: 'Created at: {createdAt}',
  },
})
