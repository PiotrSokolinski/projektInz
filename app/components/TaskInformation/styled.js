import styled from 'styled-components'

import { Colors } from 'themes'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
`

export const DescriptionContainer = styled.div``

export const Description = styled.div`
  font-size: 13px;
  color: ${Colors.black};
`

export const DescriptionTitle = styled.div`
  font-size: 15px;
  font-weight: bold;
  color: ${Colors.black};
  margin-bottom: 5px;
`

export const AuthorContainer = styled.div`
  color: ${Colors.black};
  display: flex;
  justify-content: space-between;
  padding-right: 20px;
  margin-top: 15px;
`

export const Author = styled.div`
  color: ${Colors.black};
`
