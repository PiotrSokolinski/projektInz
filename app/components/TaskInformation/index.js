/**
 *
 * TaskInformation
 *
 */

import React from 'react'
// import PropTypes from 'prop-types';
import dayjs from 'dayjs'

import { FormattedMessage } from 'react-intl'
import messages from './messages'
import * as Styled from './styled'

const TaskInformation = ({ taskData, isTask }) => (
  <Styled.Container>
    <Styled.DescriptionTitle>
      <FormattedMessage {...messages.description} />
    </Styled.DescriptionTitle>
    <Styled.Description>{taskData.description}</Styled.Description>
    <Styled.AuthorContainer>
      <FormattedMessage
        {...messages.author}
        values={{
          author: (
            <b>
              {taskData.author.firstName} {taskData.author.lastName}
            </b>
          ),
        }}
      />
      <FormattedMessage
        {...messages.createdAt}
        values={{ createdAt: <b>{dayjs(taskData.createdAt).format('DD/MM/YYYY MM:hh:mm')}</b> }}
      />
    </Styled.AuthorContainer>
  </Styled.Container>
)

TaskInformation.propTypes = {}

export default TaskInformation
