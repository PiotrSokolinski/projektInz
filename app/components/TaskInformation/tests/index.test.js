/**
 *
 * Tests for TaskInformation
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import TaskInformation from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<TaskInformation />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <TaskInformation />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <TaskInformation />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
