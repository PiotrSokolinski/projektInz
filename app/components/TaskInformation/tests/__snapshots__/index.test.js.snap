// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskInformation /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-sc-1ayn9pc-0 jagQoH"
>
  <div
    class="styled__DescriptionTitle-sc-1ayn9pc-3 eZeqmN"
  >
    <span>
      Description
    </span>
  </div>
  <div
    class="styled__Description-sc-1ayn9pc-2 fqDBCq"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to PageMaker including versions of Lorem Ipsum.
  </div>
  <div
    class="styled__AuthorContainer-sc-1ayn9pc-4 bQfJnM"
  >
    <span>
      Author: 
      <b>
        Piotr Sokolinski
      </b>
    </span>
    <span>
      Created at: 
      <b>
        27/10/2019 10:08:36
      </b>
    </span>
  </div>
</div>
`;
